src/Meilisearch/Query/SearchQueryBuilderInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Meilisearch\Query;
+
+use Meilisearch\Contracts\SearchQuery;
+
+interface SearchQueryBuilderInterface
+{
+    /**
+     * @param array<string> $facets
+     * @param array<string, mixed> $filter
+     */
+    public function build(string $indexName, string $query, array $facets, array $filter): SearchQuery;
```

Comment:
Shouldn't the `$query` actually be nullable?

Reply:
I believe it should not, the _null query_ for the search should be an empty string (Meilisearch also allows to search with an empty string, so for all of the values), and we keep it this way in the classes using the builder

Reply:
I kind of agree. I am thinking a bit ahead of time: when we will show a category page then it makes more sense that the query is nullable from the perspective of the plugin user. I know that Meilisearch needs at least an empty string...

Hmm, but it also makes sense that the $query is the first 'search argument'... I don't know... :|

Reply:
🤷 seems as a good thing to also iterate on after merging the first version of the feature 😄